File: Question_6/Java/Question6.java
import java.io.*;
import java.util.*;

class Solution {

    static String encrypt(String words) {
        String t = words.replaceAll(" ", "");
        int row = (int) Math.floor(Math.sqrt(t.length()));
        int col = (int) Math.ceil(Math.sqrt(t.length()));
        if ((row * col) < t.length()) row++;
        char[][] encrypt = new char[row][col];
        int k = 0;
        for (int i = 0; i < row; i++) {
            for (int j = 0; j < col; j++) {
                if (k < t.length()) {
                    encrypt[i][j] = t.charAt(k);
                    k++;
                }
            }
        }
        StringBuilder str = new StringBuilder();
        for (int i = 0; i < col; i++) {
            for (int j = 0; j < row; j++) {
                if ((int) (encrypt[j][i]) != 0)
                    str.append(encrypt[j][i]);
            }
            str.append(" ");
        }
        return str.toString();
    }


    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        String words = scanner.nextLine();
        scanner.close();

        String result = encrypt(words);
        // Please do not remove the below line.
        System.out.println(result);
        // Do nore print anything after this line
    }
}

File: Question_6/Python/Question6.py
import math


def encrypt(words):
    words = words.replace(" ", "")
    row = int(math.floor(math.sqrt(len(words))))
    col = int(math.ceil(math.sqrt(len(words))))
    if row * col < len(words):
        row += 1
    string = []
    k = 0
    i = 0
    while i < row * col - 1:
        if i < len(words):
            string.append(words[i])
        if i == col * (row - 1) + k:
            k += 1
            i = k
            string.append(" ")
        else:
            i += col
    return "".join(string)


def main():
    words = input()
    answer = encrypt(words)
    print(answer)


if __name__ == '__main__':
    main()
